src/FileUpload.jsx

```diff
@@ -20,6 +21,8 @@ export default function FileUpload() {
 
   const [authorized, setAuthorized] = useState(false);
 
+  const [checkingAuth, setCheckingAuth] = useState(true);
```

Comment:
checking auth is very close to the function checkauth. Lets call it checkAuthorization.
```suggestion
  const [checkingAuthorization, setCheckingAuthorization] = useState(true);
```